shared/src/main/scala/minitest/api/SourceLocation.scala

```diff
@@ -28,33 +27,18 @@ final case class SourceLocation(
 )
 
 object SourceLocation {
-  implicit def fromContext: SourceLocation =
-    SourceLocation(None, None, 0)
-  //   macro Macros.fromContext
+  implicit transparent def fromContext: SourceLocation = ~fromContextImpl(TopLevelSplice.tastyContext)
 
-  // @macrocompat.bundle class Macros(val c: whitebox.Context) {
-  //   import c.universe._
+  def fromContextImpl(implicit tasty: Tasty): Expr[SourceLocation] = {
+    val (fileName, path, line) = getSourceLocation
+    '(SourceLocation(Some(~fileName.toExpr), Some(~path.toExpr), ~line.toExpr))
+  }
 
-  //   def fromContext: Tree = {
-  //     val (fileNameExpr, pathExpr, lineExpr) = getSourceLocation
-  //     val SourceLocationSym = symbolOf[SourceLocation].companion
-  //     q"""$SourceLocationSym($fileNameExpr, $pathExpr, $lineExpr)"""
-  //   }
+  private def getSourceLocation(implicit tasty: Tasty): (String, String, Int) = {
```

Comment:
I suppose we can inline this in `fromContextImpl`. No need to maintain the keep the previous methods around

Reply:
No we cannot. The top level splice requires a call to a static method.

Reply:
This is forbidden?
```scala
implicit transparent def fromContext: SourceLocation = ~fromContextImpl(TopLevelSplice.tastyContext)

def fromContextImpl(implicit tasty: Tasty): Expr[SourceLocation] = {
  val pos = tasty.rootPosition
  val file = pos.sourceFile
  val path = file.getParent.toString
  'SourceLocation(Some(~file.getFileName.toExpr), Some(~path.toExpr), ~pos.startLine.toExpr)
}
```

Reply:
Oh, I missed. Yes, that one is possible.